pleskdistup/common/tests/actionstests.py

```diff
@@ -102,20 +117,53 @@ def test_skip_action(self):
         simple_action._prepare_action.assert_called_once()
         skip_action._prepare_action.assert_not_called()
 
+    def test_skip_based_on_saved_success_state(self):
+        simple_action = SimpleAction()
+        simple_action._prepare_action = mock.Mock(return_value=action.ActionResult())
+        saved_action = SavedAction()
+        saved_action._prepare_action = mock.Mock(return_value=action.ActionResult())
 
-class SavedAction(action.ActiveAction):
-    def __init__(self):
-        self.name = "saved"
-        self.description = "Saved action description"
+        with open("actions.json", "w") as actions_data_file:
+            actions_data_file.write('{ "actions": [ { "stage": "test_skip_based_on_saved_state", "name" : "saved", "state" : "success"}] }')
 
-    def _prepare_action(self) -> action.ActionResult:
-        return action.ActionResult()
+        with PrepareActionsFlowForTests({"test_skip_based_on_saved_state": [simple_action, saved_action]}) as flow:
+            flow.validate_actions()
+            flow.pass_actions()
 
-    def _post_action(self) -> action.ActionResult:
-        return action.ActionResult()
+        simple_action._prepare_action.assert_called_once()
+        saved_action._prepare_action.assert_not_called()
 
-    def _revert_action(self) -> action.ActionResult:
-        return action.ActionResult()
+    def test_skip_based_on_saved_state(self):
+        simple_action = SimpleAction()
+        simple_action._prepare_action = mock.Mock(return_value=action.ActionResult())
+        saved_action = SavedAction()
+        saved_action._prepare_action = mock.Mock(return_value=action.ActionResult())
+
+        with open("actions.json", "w") as actions_data_file:
+            actions_data_file.write('{ "actions": [ { "stage": "test_recheck_skip_based_on_saved_state", "name" : "saved", "state" : "skip"}] }')
+
+        with PrepareActionsFlowForTests({"test_recheck_skip_based_on_saved_state": [simple_action, saved_action]}) as flow:
+            flow.validate_actions()
+            flow.pass_actions()
+
+        simple_action._prepare_action.assert_called_once()
+        saved_action._prepare_action.assert_called_once()
+
+    def test_skip_failed_saved_state(self):
```

Comment:
These 3 tests are almost identical. Could they be converted into invocations of a single function with different parameters?

Reply:
Not sure it will simplify things, but ok let's do it